### PR TITLE
Routing

### PR DESCRIPTION
**Pull request by nhuttm**

### CODE REVIEW

File: website/client/src/components/Routes/PrivateRouteAdmin.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { IS_ADMIN } from '../../constant/constant';
+import { Redirect, Route } from 'react-router-dom';
+
+const PrivateRouteAdmin = ({ component: Component, ...rest }) =>
+    (
+        <Route {...rest} render={
+            (props) =>
+                (
+                    rest.user ? (
+                        rest.user.role === IS_ADMIN ? <Component /> : <Redirect to='/admin/login' />
```
Comment (tuandtb) [inline website/client/src/components/Routes/PrivateRouteAdmin.js]: You need to pass all props from Router to render Component, as <Component {...props} />, in order to get all router props in the child component.

File: website/client/src/components/Routes/PublicRoutes.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import NavBar from '../NavBar/NavBar';
+import Footer from '../Footer/Footer';
+import { Route } from 'react-router-dom';
+
+const PublicRoutes = ({ component: Component, ...rest }) =>
+    (
+        <>
+            <HeaderBar /> <NavBar />
+            <Route {...rest} render={ (props) =>
+                <Component />
```
Comment (tuandtb) [inline website/client/src/components/Routes/PublicRoutes.js]: Remember to pass all props to render Component.

File: website/client/src/components/Routes/PrivateRouteAdmin.js (hunk 1 above)
Comment (tuandtb) [inline website/client/src/components/Routes/PrivateRouteAdmin.js]: Where do you get "user" prop from?
A little problem with naming a variable  "IS_ADMIN", is this value is a boolean type or a string.

Comment (nhuttm) [reply website/client/src/components/Routes/PrivateRouteAdmin.js]: I get in Routes component and send to PrivateRoute as props. IS_ADMIN is a constant integer number

Comment (tuandtb) [reply website/client/src/components/Routes/PrivateRouteAdmin.js]: Ok I got it, why not connect redux in the PrivateRoute so you don't have to pass "user" every time you have new private routes.
Just name it "ADMIN", "is" or "should" is the best-practice for prefix name with a boolean variable. such as "isOpen" for the modal.

**Pull request merged**